lib/tX.ml

```diff
@@ -26,22 +24,24 @@ module Request = struct
     flags: Flags.t;
     id: int;
 
-    (** For frames split over multiple requests, first.size is the total
-        size of the frame. Each of the following requests gives the size
-        of that fragment. The receiver recovers the actual size of the
-        first fragment by subtracting all of the other sizes. *)
+    (* For frames split over multiple requests, first.size is the total
+       size of the frame. Each of the following requests gives the size
+       of that fragment. The receiver recovers the actual size of the
+       first fragment by subtracting all of the other sizes. *)
     size: int;
   }
 
-  [%%cstruct
-  type req = {
-    gref: uint32_t;
-    offset: uint16_t;
-    flags: uint16_t;
-    id: uint16_t;
-    size: uint16_t;
-  } [@@little_endian]
-  ]
+  let get_req_gref c = Cstruct.LE.get_uint32 c 0
+  let set_req_gref c gref = Cstruct.LE.set_uint32 c 0 gref
+  let get_req_offset c = Cstruct.LE.get_uint16 c 4
+  let set_req_offset c off = Cstruct.LE.set_uint16 c 4 off
+  let get_req_flags c = Cstruct.LE.get_uint16 c 6
+  let set_req_flags c flags = Cstruct.LE.set_uint16 c 6 flags
+  let get_req_id c = Cstruct.LE.get_uint16 c 8
+  let set_req_id c id = Cstruct.LE.set_uint16 c 6 id
+  let get_req_size c = Cstruct.LE.get_uint16 c 8
+  let set_req_size c size = Cstruct.LE.set_uint16 c 8 size
+  let sizeof_req = 10
```

Comment:
I'm a bit late here, but it should be `  let sizeof_req = 10`.

Reply:
oops, there are a couple of offsets wrong. sorry